cmake/sysbuild/image_signing.cmake

```diff
@@ -75,6 +75,10 @@ function(zephyr_mcuboot_tasks)
     else()
       set(imgtool_rom_command --rom-fixed @PM_MCUBOOT_PRIMARY_ADDRESS@)
     endif()
+
+    set(imgtool_directxip_hex_command --confirm)
```

Comment:
Shouldn't we only mark the image as confirmed for the `CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP_WITH_REVERT` bootloader mode?

As far as I know, the `CONFIG_MCUBOOT_BOOTLOADER_MODE_DIRECT_XIP` variant does not use the confirmation mechanism.

Reply:
Confirm should have no effect on DIRECT_XIP without revert, as the order of execution is to pick the highest slot with highest version. Confirm is only used by REVERT to mark tried image to stay.

Reply:
@de-nordic confirm - it doesn't affect that mode at all.